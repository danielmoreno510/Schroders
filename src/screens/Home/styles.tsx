import {StyleSheet} from 'react-native';
import {widthPercentageToDP} from '../../utils/responsiveUtils';

export default StyleSheet.create({
  container: {
    flex: 1,
    padding: widthPercentageToDP('5%'),
    backgroundColor: 'white',
  },

  title: {
    fontSize: 28,
    fontWeight: 'bold',
  },

  spinner: {
    marginTop: 32,
  },

  button: {
    marginTop: 40,
    marginBottom: 40,
  },

  yearContainer: {
    flexDirection: 'row',
    marginTop: 12,
  },

  year: {
    marginRight: 12,
    flex: 1,
  },
});
