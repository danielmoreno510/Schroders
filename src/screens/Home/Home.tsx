import React from 'react';
import {ScrollView, View, Text} from 'react-native';
import {Spinner} from 'native-base';

import Button from '../../components/common/Button';
import styles from './styles';
import {primaryColor} from '../../constants';
import Histogram from '../../components/common/Histogram';

const Home: React.FC<IHomeProps> = ({
  isFetchingCommitList,
  histogramData,
  year,
  getCommits,
  navigation,
}) => {
  React.useEffect(() => {
    getCommits('2023');
  }, [getCommits]);

  return (
    <ScrollView style={styles.container}>
      <Text style={styles.title}>Swift Repository</Text>
      {!isFetchingCommitList ? (
        <View>
          <Histogram histogram={histogramData} />
          <View style={styles.yearContainer}>
            {year !== '2021' && (
              <View style={styles.year}>
                <Button text="2021" onPress={() => getCommits('2021')} />
              </View>
            )}
            {year !== '2022' && (
              <View style={styles.year}>
                <Button text="2022" onPress={() => getCommits('2022')} />
              </View>
            )}
            {year !== '2023' && (
              <View style={styles.year}>
                <Button text="2023" onPress={() => getCommits('2023')} />
              </View>
            )}
          </View>
          <View style={styles.button}>
            <Button
              data-testid="list"
              text="Show List"
              onPress={() => navigation.navigate('Commit List')}
            />
          </View>
        </View>
      ) : (
        <Spinner color={primaryColor} />
      )}
    </ScrollView>
  );
};

export default Home;
