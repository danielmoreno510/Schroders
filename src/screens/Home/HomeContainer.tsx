import Home from './Home';
import {StoreService} from '../../redux/StoreService';
import {fetchCommitList} from '../../redux/actions';
import {getHistogramData} from '../../redux/selectors/commitsSelectors';

export const mapStateToProps = (state: ICommitListStateProps) => ({
  isFetchingCommitList: state.commits.isFetchingCommitList,
  commitList: state.commits.commitList,
  year: state.commits.year,
  histogramData: getHistogramData(state),
});

export const mapDispatchToProps = (dispatch: (T: ICommitReducer) => void) => ({
  getCommits: (year: string) => {
    dispatch(fetchCommitList(year));
  },
});

const HomeContainer = StoreService.connect(
  mapStateToProps,
  mapDispatchToProps,
)(Home);

export default HomeContainer;
