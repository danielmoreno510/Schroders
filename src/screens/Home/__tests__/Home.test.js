import React from 'react';
import {shallow} from 'enzyme';

import Home from '../Home';

describe('Home component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      isFetchingCommitList: false,
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Home {...testProps} />);

      it('THEN should display a regular Home', () => {
        expect(shallowWrapper.debug()).toMatchSnapshot();
      });
    });

    describe('WHEN the commits button is pressed', () => {
      const shallowWrapper = shallow(<Home {...testProps} />);
      shallowWrapper.find({'data-testid': 'list'}).simulate('press');

      it('THEN navigation function should be called', () => {
        expect(testProps.navigation.navigate).toBeCalledTimes(1);
      });
    });
  });
});
