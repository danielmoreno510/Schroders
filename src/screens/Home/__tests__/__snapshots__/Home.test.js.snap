// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Home 1`] = `
"<ScrollView style={{...}}>
  <Text style={{...}}>
    Swift Repository
  </Text>
  <View>
    <Histogram histogram={[undefined]} />
    <View style={{...}}>
      <View style={{...}}>
        <CustomButton text="2021" onPress={[Function: onPress]} />
      </View>
      <View style={{...}}>
        <CustomButton text="2022" onPress={[Function: onPress]} />
      </View>
      <View style={{...}}>
        <CustomButton text="2023" onPress={[Function: onPress]} />
      </View>
    </View>
    <View style={{...}}>
      <CustomButton data-testid="list" text="Show List" onPress={[Function: onPress]} />
    </View>
  </View>
</ScrollView>"
`;
